### ISSUE TITLE
full batch for energy training

### ISSUE THREAD

**Issue by brucefan1983** [brucefan1983/GPUMD#560]
For large training data set such as one for the periodic table, it is impossible to use a full batch. A large batch size such as 10000 is ok for force because a structure has tens of atoms on average and the actual batch size for force training is thus at least 100000, which is more than 1000/species. However, this batch size might still be too small for energy (only 100/species). To increase the accuracy of energy, it is better to independently enable the full-batch mode for energy, while still using a mini-batch for force and virial. What can we gain by using different batch sizes for energy and force? Speed of course. When force is not needed, we can omit all force-evaluation kernels, which saves about 2/3 of the computation.

**Comment by brucefan1983** [comment]
This paper shows why large batch size is needed for complex problems: https://arxiv.org/pdf/1812.06162.pdf

**Comment by brucefan1983** [comment]
Done in #562  but not only for energy, but for energy, force, and virial.
